### PR TITLE
doc: enable CI doxygen warnings verification

### PR DESCRIPTION
**Pull request by utzig**
Right now doxygen warnings are written to a separate file which is never checked by the CI. This changes the configuration to write to stderr, so that all warnings are properly checked. zboss was excluded from check since we don't have control of changes in the library.

JIRA: NCSDK-4971

-----
depends on: https://github.com/nrfconnect/sdk-nrfxlib/pull/195

### CODE REVIEW

**Comment by ru-fu** [comment]
Do you know if the CI reports errors on stderr? It doesn't report the version warnings (RemovedInSphinx30Warning: ), but I don't know if those are output to stderr or stdout.

**Comment by ru-fu** [comment]
> Well, I can add a warning back! :-) Also `RemovedInSphinx30Warning` is not a doxygen warning, but a Sphinx warning.

I know, but it's some warning that the CI doesn't catch (I _think_ it occurs on the build servers). 
My question was mostly if the change makes it even harder for us to ensure that there are no errors - right now, we can check the warnings file (even though we usually don't do that). If the CI doesn't report errors from stderr, we won't get any indication unless we build locally.
So yes, might be a good idea to add a warning back to test. ;)

**Comment by utzig** [comment]
> My question was mostly if the change makes it even harder for us to ensure that there are no errors - right now, we can check the warnings file (even though we usually don't do that). If the CI doesn't report errors from stderr, we won't get any indication unless we build locally.

Ah, now I understand your concern! This setting I changed makes warnings go to a specific file, and removing it make it go to `stderr`. What happens now is that the current build system (CMake) is already configured to make `stdout` and `stderr` go to a specific file, in doxygen's case `doxy.log`. You'll find this file inside the build directory under `nrf`, `nrfxlib`, etc, but it's currently empty exactly because the warnings were going to another file! Also those `*.log` files are archived in the build results, but I hope to do the proper checking and fail the CI when there are any.

**Comment by tejlmand** [comment]
Please rebase.

**Comment by utzig** [comment]
> Please rebase.

I could but the ordering that I would like to get the doxygen PRs in, is to first merge https://github.com/nrfconnect/sdk-nrf/pull/2460, then the relevant CI ticket (which is already approved), and afterwards this PR + its nrfxlib complimentary, then I have to finish the equivalent CI ticket over the aforementioned one.

**Comment by utzig** [comment]
> Please rebase.

Done.

File: west.yml [hunk 1]
```diff
@@ -97,7 +97,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: v1.3.0
+      revision: pull/195/head
```
Comment (carlescufi) [inline west.yml:100]: Replace with 4974ac1b3b4caa186f64341811f1844c559698c1

Comment (utzig) [reply west.yml]: Done.

**Pull request merged**